panedr/tests/test_edr.py

```diff
@@ -37,23 +48,38 @@
 EDR_DOUBLE = os.path.join(DATA_DIR, 'double.edr')
 EDR_DOUBLE_XVG = os.path.join(DATA_DIR, 'double.xvg')
 
-class TestEdrToDf(unittest.TestCase):
+EDR_Data = namedtuple('EDR_Data', ['df', 'xvgdata', 'xvgtime', 'xvgnames',
+                                   'xvgprec', 'edrfile', 'xvgfile'])
+
+@pytest.fixture(scope='module',
+                params=[(EDR, EDR_XVG),
+                        (EDR_IRREGULAR, EDR_IRREGULAR_XVG),
+                        (EDR_DOUBLE, EDR_DOUBLE_XVG)])
+def edr(request):
+    edrfile, xvgfile = request.param
+    df = panedr.edr_to_df(edrfile)
+    xvgdata, xvgnames, xvgprec = read_xvg(xvgfile)
+    xvgtime = xvgdata[:, 0]
+    xvgdata = xvgdata[:, 1:]
+    return EDR_Data(df, xvgdata, xvgtime, xvgnames, xvgprec, edrfile, xvgfile)
+    
+
+class TestEdrToDf(object):
     """
     Tests for :fun:`panedr.edr_to_df`.
     """
-    def test_output_type(self):
+    def test_output_type(self, edr):
         """
         Test that the function returns a pandas DataFrame.
         """
-        df = panedr.edr_to_df(EDR)
-        self.assertIsInstance(df, pandas.DataFrame)
+        assert isinstance(edr.df, pandas.DataFrame)
 
-    def test_columns(self):
+    def test_columns(self, edr):
         """
         Test that the column names and order match.
         """
-        df = panedr.edr_to_df(EDR)
-        ref_columns = numpy.array([u'Time', u'Bond', u'G96Angle',
+        ref_columns = numpy.insert(edr.xvgnames, 0, u'Time')
```

Comment:
Here you are mutating a module scoped fixture. It has no impact now because the attribute is only used in that test that only run once, but it can cause problem down the line. It would be safer to work with a copy.